p3a_symmetric3x3.hpp

```diff
@@ -264,6 +295,38 @@ auto frobenius_inner_product(
        a.zz() * b.zz();
 }
 
+template <class T>
+[[nodiscard]] P3A_HOST P3A_DEVICE P3A_ALWAYS_INLINE constexpr
+T determinant(symmetric3x3<T> const& a)
+{
+  return
+    a.xx() * a.yy() * a.zz() -
+    a.xx() * a.yz() * a.yz() -
+    a.xy() * a.xy() * a.zz() +
+    T(2.0) * a.xy() * a.xz() * a.yz() -
+    a.xz() * a.xz() * a.yy();
+}
+
+template <class T>
+[[nodiscard]] P3A_HOST P3A_DEVICE P3A_ALWAYS_INLINE constexpr
+auto inverse(symmetric3x3<T> const& a)
+{
+  auto const det = determinant(a);
+  T constexpr epsilon = std::numeric_limits<T>::epsilon();
+  if (std::abs(det) <= epsilon)
+  {
+    throw std::logic_error("non-invertible matrix");
+  }
```

Comment:
Right now, other `inverse` methods do not check for zero determinants. For now I am personally okay having none of them check, because in some cases we assume the matrix is invertible and checking costs time (and is not vectorized). If we were to have a function that checks this, in order to run on GPUs it needs to use a return code like the `enum class errc` approach.

Reply:
thanks @ibaned.  I guarded the check behind `P3A_DEBUG` - feel free to zap or change to a more descriptive name.  Error checks in material models are many times necessary, I think it would be nice to have one way to include debug code and extra error checking that could be shut off on the GPU.